src/control.lua

```diff
@@ -568,11 +589,16 @@ function HandleInputTank(entityData)
 			if isFluidLegal(fluid.name) then
 				if fluid.amount > 1 then
 					local fluid_taken = math.ceil(fluid.amount) - 1
-					AddItemToInputList(fluid.name, fluid_taken)
+					AddItemToInputList(fluid.name, fluid_taken, "normal")
 					fluid.amount = fluid.amount - fluid_taken
 					fluidbox[1] = fluid
 				else
-					if entity.get_merged_signal({name="signal-P",type="virtual"}) == 1 then
+					local signal = entity.get_signal(
+						{name = "signal-P", type = "virtual"},
+						defines.wire_connector_id.circuit_red,
+						defines.wire_connector_id.circuit_green
+					)
```

Comment:
`get_signal` does not exist on LuaEntity in 1.1.110. A version check is needed here.